src/script/cpatch.py

```diff
@@ -0,0 +1,727 @@
+        return self._cli.root_build
+
+    @property
+    def build_dir(self):
+        if self._cli.build_dir:
+            return pathlib.Path(self._cli.build_dir).absolute()
+        return pathlib.Path(".").absolute()
+
+    @property
+    def source_dir(self):
+        if self._cli.source_dir:
+            return pathlib.Path(self._cli.source_dir).absolute()
+        return (self.build_dir / "..").absolute()
+
+    @property
+    def push(self):
+        return self._cli.push
+
+    @property
+    def pull(self):
+        return self._cli.pull
+
+    @property
+    def strip_binaries(self):
+        return self._cli.strip
+
+    @property
+    def components_selected(self):
+        return bool(self._cli.components)
+
+    def build_components(self):
+        if self._cli.components:
+            return self._cli.components
+        # because everything is nothing.
+        return list(Component)
+
+    def setup_logging(self):
+        logging.config.dictConfig(
+            {
+                "version": 1,
+                "disable_existing_loggers": True,
+                "formatters": {
+                    "default": {
+                        "class": "logging.Formatter",
+                        "format": "%(asctime)s %(levelname)s: %(message)s",
+                    },
+                },
+                "handlers": {
+                    "stderr": {
+                        "formatter": "default",
+                        "level": "DEBUG",
+                        "class": "logging.StreamHandler",
+                        "stream": sys.stderr,
+                    }
+                },
+                "root": {
+                    "level": self._cli.log_level,
+                    "handlers": ["stderr"],
+                },
+            }
+        )
+
+    @classmethod
+    def parse(cls):
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "--base",
+            default=BASE_IMAGE,
+            help=f"base container image [default: {BASE_IMAGE}]",
+        )
+        parser.add_argument(
+            "--target",
+            default=DEST_IMAGE,
+            help=f"target image to create [default: {DEST_IMAGE}]",
+        )
+        parser.add_argument(
+            "--push", action="store_true", help="push when done"
+        )
+        parser.add_argument(
+            "--pull",
+            action="store_true",
+            help=(
+                "pull base image before building (useful to prevent building"
+                " with a stale base image)"
+            ),
+        )
+        parser.add_argument(
+            "--build-dir",
+            "-b",
+            help="path to the build dir (defaults to current workding dir)",
+        )
+        parser.add_argument(
+            "--source-dir",
+            help="path to the source dir (defaults to parent of build dir)",
+        )
+        parser.add_argument(
+            "--strip", action="store_true", help="strip binaries"
```

Comment:
Do you know if there were there particular cases for wanting vs. not wanting this or was this just something taken from the other cpatch?

Reply:
Just taken from the old cpatch. I know that if you plan on using a debugger you generally don't want to strip the binaries. I'm not sure why it lets you strip the binaries while running cpatch, but it was there.